### ISSUE TITLE
[Event Request] Table 36 Sales Header - OnValidateSellToContactNoOnBeforeCheckContactNo

### ISSUE THREAD

**Issue by fvanderhek** [microsoft/ALAppExtensions#7046]
Requesting a new Event Publisher on Table 36 Sales Header

Event
`OnValidateSellToContactNoOnBeforeCheckContactNo(ContBusinessRelation : Record "Contact Business Relation",Cont : Record Contact,VAR IsHandled : Boolean)`

**Code**
## Sell-to Contact No. - OnValidate()
```
TestStatusOpen;

IF "Sell-to Contact No." <> '' THEN
  IF Cont.GET("Sell-to Contact No.") THEN
    Cont.CheckIfPrivacyBlockedGeneric;

IF ("Sell-to Contact No." <> xRec."Sell-to Contact No.") AND
   (xRec."Sell-to Contact No." <> '')
THEN BEGIN
  IF ("Sell-to Contact No." = '') AND ("Opportunity No." <> '') THEN
    ERROR(Text049,FIELDCAPTION("Sell-to Contact No."));
  IF GetHideValidationDialog OR NOT GUIALLOWED THEN
    Confirmed := TRUE
  ELSE
    Confirmed := CONFIRM(ConfirmChangeQst,FALSE,FIELDCAPTION("Sell-to Contact No."));
  IF Confirmed THEN BEGIN
    IF InitFromContact("Sell-to Contact No.","Sell-to Customer No.",FIELDCAPTION("Sell-to Contact No.")) THEN
      EXIT;
    IF "Opportunity No." <> '' THEN BEGIN
      Opportunity.GET("Opportunity No.");
      IF Opportunity."Contact No." <> "Sell-to Contact No." THEN BEGIN
        MODIFY;
        Opportunity.VALIDATE("Contact No.","Sell-to Contact No.");
        Opportunity.MODIFY;
      END
    END;
  END ELSE BEGIN
    Rec := xRec;
    EXIT;
  END;
END;

IF ("Sell-to Customer No." <> '') AND ("Sell-to Contact No." <> '') THEN BEGIN
  Cont.GET("Sell-to Contact No.");
  IF ContBusinessRelation.FindByRelation(ContBusinessRelation."Link to Table"::Customer,"Sell-to Customer No.") THEN;
```
**>>> New Code**
```
    IsHandled := FALSE;
    OnValidateSellToContactNoOnBeforeCheckContactNo(ContBusinessRelation,Cont,IsHandled);
    IF NOT IsHandled THEN
```
**<<< New Code End**
```
      IF (ContBusinessRelation."Contact No." <> Cont."Company No.") AND (ContBusinessRelation."Contact No." <> Cont."No.") THEN
        ERROR(Text038,Cont."No.",Cont.Name,"Sell-to Customer No.");
END;

IF "Sell-to Contact No." <> '' THEN
  IF Cont.GET("Sell-to Contact No.") THEN
    IF ("Salesperson Code" = '') AND (Cont."Salesperson Code" <> '') THEN
      VALIDATE("Salesperson Code",Cont."Salesperson Code");

UpdateSellToCust("Sell-to Contact No.");
UpdateSellToCustTemplateCode;
UpdateShipToContact;
```

In Our Solution we have contacts that are not linked to a company but can also become customers that can order items and so on. We need to be able to skip the default error here to be able to use these kind of customers. Can you please add this publisher?

**Comment by fvanderhek** [comment]
I disagree, This Publisher is requested for Sell-To Contact No, not Bill-To Contact No, Please review.
#6866 is related to Bill-To Contact No.

**Comment by StanPesotskiy** [comment]
What I see now in Sales Header table:

```
                if ("Bill-to Customer No." <> '') and ("Bill-to Contact No." <> '') then
                    CheckContactRelatedToCustomerCompany("Bill-to Contact No.", "Bill-to Customer No.", CurrFieldNo);
```
and 
```
                if ("Sell-to Customer No." <> '') and ("Sell-to Contact No." <> '') then
                    CheckContactRelatedToCustomerCompany("Sell-to Contact No.", "Sell-to Customer No.", CurrFieldNo);
```
where 

```
local procedure CheckContactRelatedToCustomerCompany(ContactNo: Code[20]; CustomerNo: Code[20]; CurrFieldNo: Integer);
    var
        Contact: Record Contact;
        ContBusRel: Record "Contact Business Relation";
        IsHandled: Boolean;
    begin
        IsHandled := false;
        OnBeforeCheckContactRelatedToCustomerCompany(Rec, CurrFieldNo, IsHandled);
        if IsHandled then
            exit;

        Contact.Get(ContactNo);
        if ContBusRel.FindByRelation(ContBusRel."Link to Table"::Customer, CustomerNo) then
            if (ContBusRel."Contact No." <> Contact."Company No.") and (ContBusRel."Contact No." <> Contact."No.") then
                Error(Text038, Contact."No.", Contact.Name, CustomerNo);
    end;
```
Subscribing to `OnBeforeCheckContactRelatedToCustomerCompany()` you can avoid Error(Text038...) for both "Sell-to Contact No." and "Bill-to Contact No.".

**Comment by fvanderhek** [comment]
In which version of BC is this, when i review BC16 i cannot find this function. I'm currently refactoring in Spring Release to go to VS Code for our vertical solution.

**Comment by StanPesotskiy** [comment]
It was done 3 weeks ago and is shipped with version 16.2.
